nuxeo-drive-client/nxdrive/osi/windows/windows.py

```diff
@@ -143,19 +185,49 @@ def register_protocol_handlers(self):
             [('', _winreg.REG_SZ, command)],
         )
 
-    def _recursive_delete(self, reg, start_path, end_path):
-        try:
-            while (len(start_path) < len(end_path)):
-                _winreg.DeleteKey(reg, end_path)
-                end_path = end_path[0:end_path.rfind('\\')]
-        except Exception, e:
-            pass
+    def traverse_path(self, reg, start_path, access_mask, enrties_list):
+        key = _winreg.OpenKey(reg, start_path, 0, access_mask)
+        count = 0
+        child_count = _winreg.QueryInfoKey(key)[0]
+        while count < child_count:
+            reg_name = _winreg.EnumKey(key, count)
+            self.traverse_path(reg, start_path + '\\' + reg_name, access_mask, enrties_list)
+            enrties_list.append(start_path + '\\' + reg_name)
+            count = count + 1
+        _winreg.CloseKey(key)
+
+    '''
+        delete registry entries added for windows 10 navigation panel
+    '''
+
+    def _delete_registry_entries(self):
+        reg = _winreg.ConnectRegistry(None, _winreg.HKEY_CURRENT_USER)
+        path = 'Software\\Classes\\CLSID'
+        self._recursive_delete(reg, path, self.guid)
+        self._recursive_delete(reg, self.EXPLORER + '\Desktop\NameSpace', self.guid)
+        self._delete_reg_value(reg, self.EXPLORER + '\HideDesktopIcons\NewStartPanel', self.guid)
+
+    def _recursive_delete(self, reg, start_path, name):
+        if platform.machine().endswith('64'):
+            access_mask = _winreg.KEY_ALL_ACCESS | _winreg.KEY_WOW64_64KEY
+        else:
+            access_mask = _winreg.KEY_ALL_ACCESS
+
+        reg = _winreg.ConnectRegistry(None, _winreg.HKEY_CURRENT_USER)
+        entries_list = list()
+        self.traverse_path(reg, start_path + '\\' + name, access_mask, entries_list)
+        entries_list.append(start_path + '\\' + name)
+        key = _winreg.OpenKey(reg, start_path, 0, access_mask)
+        for entry in entries_list:
+            local_path = entry
+            _winreg.DeleteKey(key, local_path.replace(start_path + '\\', ''))
+        _winreg.CloseKey(key)
 
     def unregister_protocol_handlers(self):
         app_name = self._manager.get_appname()
         reg = _winreg.ConnectRegistry(None, _winreg.HKEY_CURRENT_USER)
-        self._recursive_delete(reg, 'Software\\', 'Software\\' + app_name + '\\Protocols\\nxdrive\\shell\\open\\command')
-        self._recursive_delete(reg, 'Software\\Classes\\', 'Software\\Classes\\nxdrive\\shell\\open\\command')
+        self._recursive_delete(reg, 'Software', app_name)
```

Comment:
Please help me to understand.
I believe earlier version function and newly added function, both cleanup same set of registry entries. I am not sure, how the new registry clean function to be introduced.

Currently two registry entries are made, which are added on installation and removed on uninstall.
I am not aware of entries, which are not related on protocal handlers in this path
[1] SOFTWARE\Nuxeo Drive\Protocols\nxdrive\shell\open\command
[2] SOFTWARE\Classes\nxdrive\shell\open\command